pong.py

```diff
@@ -23,31 +25,53 @@ def __init__(self, mlhost, mlport, screen_width, screen_height):
         self.clock = pymlgame.Clock(15)
         self.gameover = False
         self.players = {}
-        self.state = "WAITING"
+        self.state = "WAITING_FOR_PLAYERS"
         self.init_ball()
 
     def init_ball(self):
         ball_position = Point(self.screen.width // 2, self.screen.height // 2)
         ball_velocity = Point(random.choice([-1, 1]) * 0.5, random.choice([-1, 1]) * 0.2)
         self.ball = Ball(ball_position, ball_velocity)
-        self.ball_surface = pymlgame.Surface(1, 1)
-        self.ball_surface.draw_dot((0, 0), pymlgame.GREEN)
 
-    def update(self):
-        """
-        Update the screens contents in every loop.
-        """
-        if self.state == "WAITING":
-            pass
+    def update_and_render(self):
+        self.screen.reset()
+
+        if self.state == "WAITING_FOR_PLAYERS":
+            self.update_waiting()
+            self.render_waiting()
         elif self.state == "START_GAME":
             self.update_start_game()
         elif self.state == "RUNNING":
             self.update_running()
+            self.render_running()
         elif self.state == "GAME_OVER":
             self.update_game_over()
         else:
             print("That's simply poor programming...")
 
+        self.screen.update()
+        self.clock.tick()
+
+    def update_waiting(self):
+        if len(self.players) == 2:
+            self.state = "START_GAME"
+
+    def render_waiting(self):
+        p0_color = pymlgame.GREY5
```

Comment:
set either grey, or green, rather than overwriting it. use inline `if..else` or ternary.